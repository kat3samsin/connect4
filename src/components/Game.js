import React, { Component } from 'react';
import { connect } from 'react-redux';

import Board from './Board';
import { initialize } from '../actions/GameActions';

export class Game extends Component {
  initializeBoard() {
    this.props.init();
  }
  render() {
    return (
      <div>
        <button className="btn" onClick={this.initializeBoard.bind(this)}>
          New Game
        </button>
        {this.props.board.length > 0 ? (
          <div>
            {this.props.winner ? (
              <h2>Game Over! Player {this.props.winner} wins!</h2>
            ) : (
              <h2>Player {this.props.player}</h2>
            )}
            <Board board={this.props.board} />
          </div>
        ) : null}
      </div>
    );
  }
}
const mapStateToProps = state => {
  return state;
};
function mapDispatchToProps(dispatch) {
  return {
    init: () => {
      dispatch(initialize());
    }
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Game);
