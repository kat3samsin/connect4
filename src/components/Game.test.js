import React from 'react';
import { shallow } from 'enzyme';
import { Game } from './Game';

describe('Game', () => {
  let wrapper;
  let initFn;

  beforeEach(() => {
    initFn = jest.fn();
    wrapper = shallow(
      <Game
        board={[[0, 0, 0], [0, 0, 0], [0, 0, 0]]}
        init={initFn}
        player={1}
      />
    );
  });

  afterEach(() => {
    initFn.mockReset();
  });

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a button', () => {
    expect(wrapper.find('button')).toHaveLength(1);
  });

  it('clicks new game', () => {
    wrapper.find('button').simulate('click');
    expect(initFn).toBeCalled();
  });

  it('renders a Board', () => {
    expect(wrapper.find('Board')).toHaveLength(1);
  });

  it('has winner > should display > Game Over! Player 2 wins', () => {
    wrapper = shallow(
      <Game
        board={[[0, 0, 0], [0, 0, 0], [0, 0, 0]]}
        init={initFn}
        winner={2}
      />
    );
    expect(wrapper.find('h2').text()).toEqual('Game Over! Player 2 wins!');
  });

  it('no winners yet > should display > Player 1', () => {
    wrapper = shallow(
      <Game
        board={[[0, 0, 0], [0, 0, 0], [0, 0, 0]]}
        init={initFn}
        player={1}
      />
    );
    expect(wrapper.find('h2').text()).toEqual('Player 1');
  });
});
