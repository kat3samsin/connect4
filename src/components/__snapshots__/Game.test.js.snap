// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders without crashing 1`] = `ShallowWrapper {}`;
